pkg/operator/encryption/key_controller_test.go

```diff
@@ -128,37 +128,48 @@ func TestKeyController(t *testing.T) {
 			},
 		},
 
-		// Verifies if a new key is not created when there is a valid write key in the system.
 		{
-			name: "no-op when a valid write key exists",
+			name: "no-op when a valid write key exists, but is not migrated",
 			targetGRs: []schema.GroupResource{
 				{Group: "", Resource: "secrets"},
 			},
 			initialObjects: []runtime.Object{
 				createDummyKubeAPIPod("kube-apiserver", "kms"),
-				createEncryptionKeySecretWithRawKey("kms", []schema.GroupResource{{Group: "", Resource: "secrets"}}, 7, []byte("61def964fb967f5d7c44a2af8dab6865")),
+				createEncryptionKeySecretWithRawKey("kms", nil, 7, []byte("61def964fb967f5d7c44a2af8dab6865")),
 			},
 			apiServerObjects: apiServerAesCBC,
 			targetNamespace:  "kms",
 			expectedActions:  []string{"list:pods:kms", "get:secrets:kms", "list:secrets:openshift-config-managed"},
 		},
 
-		// Checks if a new key is not created when there is a valid write (migrated/used) key in the system.
 		{
-			name: "no-op when a valid migrated key exists",
+			name: "no-op when a valid write key exists, is migrated, but not expired",
 			targetGRs: []schema.GroupResource{
 				{Group: "", Resource: "secrets"},
 			},
 			initialObjects: []runtime.Object{
 				createDummyKubeAPIPod("kube-apiserver", "kms"),
-				createMigratedEncryptionKeySecretWithRawKey("kms", []schema.GroupResource{{Group: "", Resource: "secrets"}}, 3, []byte("61def964fb967f5d7c44a2af8dab6865"), time.Now()),
+				createMigratedEncryptionKeySecretWithRawKey("kms", []schema.GroupResource{{Group: "", Resource: "secrets"}}, 7, []byte("61def964fb967f5d7c44a2af8dab6865"), time.Now()),
 			},
 			apiServerObjects: apiServerAesCBC,
 			targetNamespace:  "kms",
 			expectedActions:  []string{"list:pods:kms", "get:secrets:kms", "list:secrets:openshift-config-managed"},
 		},
 
-		// Checks if a new write key is created because the previous one was migrated.
+		{
+			name: "creates a new write key because previous one is migrated, but has no migration timestamp",
```

Comment:
okay but if a secret doesn't have a migration timestamp then we dead lock.

Reply:
no, there is an extra test case. If there is no timestamp, we will behave as it is expired.